src/lib-imap/imap-capability-list.h

```diff
@@ -0,0 +1,81 @@
+#ifndef IMAP_CAPABILITY_LIST_H
+#define IMAP_CAPABILITY_LIST_H
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include "lib.h"
+#include "array.h"
+
+/* bitflags to represent different conditions of
+   the current imap session */
+
+#define CAP_NEVER       0        /* never show this capability */
+#define CAP_PREAUTH     (1 << 0) /* show this cap before login */
```

Comment:
Using an enum instead of #defines makes gdb/other debuggers pretty-print the values.